tests/test_optimization.py

```diff
@@ -0,0 +1,37 @@
+# Test invariants assumed for certain optimizations
```

Comment:
please move this test file to `tests/core/test_immutable_tile_property.py` and rename accordingly